packages/integration-tests/src/utils.rs

```diff
@@ -176,8 +177,15 @@ pub fn proper_instantiate() -> (App, CwTemplateContract) {
 }
 
 pub fn get_dr_id(res: AppResponse) -> String {
-    // TODO: this is an ugly way to get the dr_id.
-    // although PostDataRequest on the DataRequest contract returns it in `data`, the Proxy contract does not yet.
-    // https://github.com/sedaprotocol/seda-chain-contracts/issues/68
-    res.events.last().unwrap().attributes[2].value.clone()
+    let binary = res.data.unwrap();
+
+    // I don't know why protobuf adds 2 bytes to the beginning of the data payload but it does
+    let binary = &binary.to_vec()[2..];
```

Comment:
This feels fragile? The examples linked in the docs don't seem to do anything like this, maybe we should follow those and not use the utility?

Reply:
I would like to know what are these 2 bytes (always , `[10,68]`)

Reply:
If not mistaken, this comes from here, right?

`packages/data-requests/src/data_request.rs`, line 113:
```
            .set_data(to_binary(&posted_dr.dr_id)?)
```

The thing is that `dr_id` of type `Hash` which is an alias type of `String`.
So in reality, we are representing bytes as a string, and then we are sending those bytes as if they were strings.  
Also, as a side note, `Binary` does expect bytes or base64-encoded strings. 😕

Maybe knowing that `dr_id` are indeed bytes, we should take that into consideration for its serialilzation and, even, benefit from it.

Here you have a snippet of a simple test to show what I mean:

```rust
#[test]
fn test_binary_encoding() {
    // Hash example (sha256 of "data", not that it really matters!)
    let hash_string = "3a6eb0790f39ac87c94f3856b2dd2c5d110e6811602261a9a923d3bb23adc8b7";
    // Bytes
    let hash_bytes = hex::decode(hash_string).unwrap();
    // Encoded oofline to avoid adding a  new dependency
    let hash_base64 =
        "M2E2ZWIwNzkwZjM5YWM4N2M5NGYzODU2YjJkZDJjNWQxMTBlNjgxMTYwMjI2MWE5YTkyM2QzYmIyM2FkYzhiNw==";

    // Simple string
    let t1 = to_binary(hash_string).unwrap();
    // 
    // data in base64
    let t2 = cosmwasm_std::Binary::from_base64(hash_base64).unwrap();
    let t3 = cosmwasm_std::Binary::from(hash_bytes);

    println!("binary 1: {}", hex::encode(&t1));
    println!("binary 2: {}", hex::encode(&t2));
    println!("binary 3: {}", hex::encode(&t3));

    println!(
        "recovered text 1: {}",
        String::from_utf8(t1.to_vec()).unwrap()
    );
    println!(
        "recovered text 2: {}",
        String::from_utf8(t2.to_vec()).unwrap()
    );
    println!("recovered text 2 base64: {}", t2);
    println!("recovered text 3: {}", hex::encode(t3));
}
```

The output is as follows:
```
binary 1: 223361366562303739306633396163383763393466333835366232646432633564313130653638313136303232363161396139323364336262323361646338623722
binary 2: 33613665623037393066333961633837633934663338353662326464326335643131306536383131363032323631613961393233643362623233616463386237
binary 3: 3a6eb0790f39ac87c94f3856b2dd2c5d110e6811602261a9a923d3bb23adc8b7
recovered text 1: "3a6eb0790f39ac87c94f3856b2dd2c5d110e6811602261a9a923d3bb23adc8b7"
recovered text 2: 3a6eb0790f39ac87c94f3856b2dd2c5d110e6811602261a9a923d3bb23adc8b7
recovered text 2 base64: M2E2ZWIwNzkwZjM5YWM4N2M5NGYzODU2YjJkZDJjNWQxMTBlNjgxMTYwMjI2MWE5YTkyM2QzYmIyM2FkYzhiNw==
recovered text 3: 3a6eb0790f39ac87c94f3856b2dd2c5d110e6811602261a9a923d3bb23adc8b7
```

As you may see, it is more efficient to serialize and deserialize as bytes.